src/message/messageActionSheet.js

```diff
@@ -233,14 +233,21 @@ export const showActionSheet = (
     ? constructHeaderActionButtons(params)
     : constructMessageActionButtons(params);
   const callback = buttonIndex => {
-    allButtons[optionCodes[buttonIndex]]({
-      dispatch,
-      subscriptions: params.backgroundData.subscriptions,
-      auth: params.backgroundData.auth,
-      ownEmail: params.backgroundData.ownEmail,
-      _,
-      ...params,
-    });
+    (async () => {
+      try {
+        await allButtons[optionCodes[buttonIndex]]({
+          dispatch,
+          subscriptions: params.backgroundData.subscriptions,
+          auth: params.backgroundData.auth,
+          ownEmail: params.backgroundData.ownEmail,
+          _,
+          ...params,
+        });
+      } catch (err) {
+        const titleString = allButtons[optionCodes[buttonIndex]].title;
```

Comment:
It's harmless right now, but it's probably better to cache `allButtons[optionCodes[buttonIndex]]` before the `await` is performed.

(`allButtons` *shouldn't* ever be modified – but if it is, we might end up accessing different values of `allButtons` before and after the `await`, if two actions are performed in quick succession before the first one fails.)

Reply:
Doesn't the callback have a closure over `allButtons[optionCodes[buttonIndex]]`, so that even if another action is performed successively, the previous callback isn't affected?

Reply:
Not being used in the latest change at all, so haven't cached it.